scraping/scraping/start_scripts/amazon_eu.py

```diff
@@ -0,0 +1,278 @@
+import json
+import pkgutil
+
+# from bisect import bisect_left
+from typing import Any, List, Optional, Tuple
+
+from core.domain import ConsumerLifestageType, GenderType
+
+
+# use bisect.bisect_left if we ever update to 3.10
+# on python 3.9 bisect_left lacks the `key` argument
+def bisect_left(a: list, x: Any, key: Any) -> int:
+    low = 0
+    high = len(a)
+    while low < high:
+        mid = (low + high) // 2
+        if key(a[mid]) < x:
+            low = mid + 1
+        else:
+            high = mid
+    return low
+
+
+def read_json(path: str) -> list:
+    data = pkgutil.get_data("scraping", path)
+    assert data
+    return json.loads(data.decode("utf-8"))
+
+
+sort_key = "{root}{path}/".format_map
+browse_tree_leaves = sorted(read_json("data/amazon_eu_browse_nodes.json"), key=sort_key)
+
+# unfortunately the json file only contains leaf nodes
+# so we have to do some extra work to map a browse node path to its leaf nodes.
+# we wouldnt need to do that if we had a list of internal nodes aswell.
+
+
+def find_leaf_nodes_by_path(path: str) -> Tuple[int, int]:
+    """
+    find all leaf nodes matching a given path
+
+    Args:
+        path (str): the name of some amazon browse node
+
+    Returns:
+        Tuple[int, int]: the range of leaf nodes given as (length, offset)
+    """
+    seperator = "/"
+    assert not path.startswith(seperator) and not path.endswith(seperator)
+    one_after_seperator = chr(ord(seperator) + 1)
+    start = bisect_left(browse_tree_leaves, path + seperator, key=sort_key)
+    end = bisect_left(browse_tree_leaves, path + one_after_seperator, key=sort_key)
```

Comment:
Is it really necessary using the bisect algorithm here? Normal list would cause real performance problems? 

Implementing `bisect_left` above does not really seems to be a good idea because it is part of the std-lib.

Reply:
We should not wait for #76 it has a dependency on https://github.com/EasyPi/docker-scrapyd/issues/2.

As discussed offline, I would definitely prefer avoiding using `bisect` at all and use use some other data structure and code that is more readable (and maintainable).